### PR TITLE
PCBS and Parquet support ANSI year month interval type [databricks]

### LINKED ISSUE #2814: SPARK-35243 has introduced support for ANSI interval types
Add cache support for ANSI Interval types as in SPARK-35243

### PR DESCRIPTION
**Pull request by res-life**
Closes #2814 

1. Parquet support ANSI year month interval type
2. PCBS support ANSI year month interval type

Signed-off-by: Chong Gao <res_life@163.com>

### CODE REVIEW

**Comment by res-life** [comment]
build

**Comment by res-life** [comment]
build

**Comment by res-life** [comment]
build

**Comment by res-life** [comment]
Should re-target branch-22.06, convert it to draft.

File: tests/src/test/330/scala/com/nvidia/spark/rapids/IntervalSuite.scala [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+import java.io.File
+import java.time.{Duration, Period}
+
+import org.apache.spark.sql.{DataFrame, SparkSession}
+
+/**
+ * Can not put this suite to cache_test.py because of currently Pyspark not have year-month type.
+ * See: https://github.com/apache/spark/blob/branch-3.3/python/pyspark/sql/types.py
+ */
+class IntervalSuite extends SparkQueryCompareTestSuite {
```
Comment (firestarman) [inline tests/src/test/330/scala/com/nvidia/spark/rapids/IntervalSuite.scala:27]: Need to skip these tests for Spark before 3.3.0

File: tests/src/test/330/scala/com/nvidia/spark/rapids/IntervalSuite.scala [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+  /**
+   * Set environment variable: "spark.sql.cache.serializer"
+   * "spark.sql.cache.serializer" is a static SQL configuration, can NOT set/unset it.
+   * Should specify it as environment variable when running the MVN, e.g:
+   * SPARK_CONF="spark.sql.cache.serializer=com.nvidia.spark.ParquetCachedBatchSerializer" mvn test
+   * This method is a hack code to set environment variable in the test code. it's a workaround.
+   * Should move this test suite to cache_test.py after Pyspark supports year-month type
+   */
+  private def setPCBS() = {
```
Comment (firestarman) [inline tests/src/test/330/scala/com/nvidia/spark/rapids/IntervalSuite.scala:37]: This is not recommended as you may know. Instead, [you can control the spark session creation by setting this static config or not here](https://github.com/NVIDIA/spark-rapids/blob/branch-22.04/tests/src/test/scala/com/nvidia/spark/rapids/SparkQueryCompareTestSuite.scala#L68).

File: tests/src/test/330/scala/com/nvidia/spark/rapids/IntervalSuite.scala (hunk 1 above)
Comment (res-life) [reply tests/src/test/330/scala/com/nvidia/spark/rapids/IntervalSuite.scala:27]: It's in `test/330` directory and will skip for Spark before 3.3.0

File: tests/src/test/330/scala/com/nvidia/spark/rapids/IntervalSuite.scala (hunk 2 above)
Comment (res-life) [reply tests/src/test/330/scala/com/nvidia/spark/rapids/IntervalSuite.scala:37]: This is temp and will be removed after Pyspark supports year month interval, so i would suggest not change the spark session creation for this.

**Comment by res-life** [comment]
build

**Comment by res-life** [comment]
@razajafri  Help review, thanks.

**Comment by firestarman** [comment]
LGTM

**Comment by razajafri** [comment]
@res-life why are we choosing to write scalatest when as a convention we write python tests? It makes sense to do scala tests for situations where writing a python test isn't possible e.g. writing a unit test for a private method or may be you have to inject variables into a class before testing. We also use python tests in environments where it's not possible for us to run scala tests.

**Comment by res-life** [comment]
@razajafri 
Currently Pyspark does not have year month type, please see
https://github.com/apache/spark/blob/branch-3.3/python/pyspark/sql/types.py
Day time type is supported in 
[SPARK-37279 Support DayTimeIntervalType in createDataFrame, collect and Python UDF](https://github.com/apache/spark/pull/34614)
Is waiting the Spark to support year month type.

**Review by razajafri — APPROVED** [review]

**Pull request merged**